### PR TITLE
Add enrollment logic

### PR DESCRIPTION
**Pull request by abhishekshree**

### CODE REVIEW

File: company/db.hr.go [hunk 1]
```diff
@@ -17,6 +17,11 @@ func deleteHR(ctx *gin.Context, id uint) error {
 	return tx.Error
 }
 
+func updateHR(ctx *gin.Context, cid uint, req *updateHRRequest) error {
+	tx := db.WithContext(ctx).Model(&CompanyHR{}).Where("company_id = ?", cid).Updates(req)
```
Comment (1-Harshit) [inline company/db.hr.go:21]:
```suggestion
	tx := db.WithContext(ctx).Model(&CompanyHR{}).Where("company_id = ? AND id=?", cid, req.ID).Updates(req)
```

File: company/company.hr.go [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+	if addHRRequest.CompanyID != 0 {
+		ctx.AbortWithStatusJSON(http.StatusBadRequest, gin.H{"error": "Company ID is not allowed"})
+		return
+	}
+
+	// companyID, err := rc.ExtractCompanyID(ctx)
+	user_email := middleware.GetUserID(ctx)
+	companyID, err := FetchCompanyIDByEmail(ctx, user_email)
+
```
Comment (1-Harshit) [inline company/company.hr.go:26]: remove this line

File: company/company.hr.go [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+	if err != nil {
+		ctx.AbortWithStatusJSON(http.StatusBadRequest, gin.H{"error": err.Error()})
+		return
+	}
+
+	addHRRequest.CompanyID = companyID
+
+	err = addHR(ctx, &addHRRequest)
+
```
Comment (1-Harshit) [inline company/company.hr.go:35]: remove line

File: company/db.hr.go (hunk 1 above)
Comment (abhishekshree) [reply company/db.hr.go:21]: done

File: company/company.hr.go [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+}
+
+type updateHRRequest struct {
+	Name        string `json:"name"`
+	Phone       string `json:"phone"`
+	Designation string `json:"designation"`
+}
+
+func putHRHandler(ctx *gin.Context) {
```
Comment (1-Harshit) [inline company/company.hr.go:51]: extract hrid
bsdk

File: rc/company.rc.go [hunk 5]
```diff
@@ -35,6 +35,55 @@ func getCompanyRCHandler(ctx *gin.Context) {
...
+	for _, rc := range allRcs {
+		for _, enrolledRC := range enrolledRcs {
+			if rc.ID == enrolledRC.ID {
+				companyAllRCResponse = append(companyAllRCResponse, getAllRCHandlerForCompanyResponse{
+					ID:       rc.ID,
+					Name:     string(rc.Type) + " " + rc.AcademicYear,
+					Enrolled: 1,
+				})
+			} else {
+				companyAllRCResponse = append(companyAllRCResponse, getAllRCHandlerForCompanyResponse{
```
Comment (1-Harshit) [inline rc/company.rc.go:75]: Bruh redundent reponse wtf
response will be:
1, intern, 1
2, intern, 0
1, placement, 0
2, placement, 0

**Review by 1-Harshit — CHANGES_REQUESTED** [review]

File: company/company.hr.go [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+		ctx.AbortWithStatusJSON(http.StatusBadRequest, gin.H{"error": err.Error()})
+		return
+	}
+
+	if addHRRequest.CompanyID != 0 {
+		ctx.AbortWithStatusJSON(http.StatusBadRequest, gin.H{"error": "Company ID is not allowed"})
+		return
+	}
+
+	// companyID, err := rc.ExtractCompanyID(ctx)
```
Comment (1-Harshit) [inline company/company.hr.go:23]: why not?

File: company/company.hr.go [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+
+func putHRHandler(ctx *gin.Context) {
+	var req updateHRRequest
+
+	if err := ctx.ShouldBindJSON(&req); err != nil {
+		ctx.AbortWithStatusJSON(http.StatusBadRequest, gin.H{"error": err.Error()})
+		return
+	}
+
+	hrid := middleware.GetUserID(ctx)
```
Comment (1-Harshit) [inline company/company.hr.go:57]: sir, path variable :)

Comment (1-Harshit) [reply company/company.hr.go:57]: make sure id==0

File: rc/company.rc.go [hunk 8]
```diff
@@ -35,6 +35,60 @@ func getCompanyRCHandler(ctx *gin.Context) {
...
+	}
+
+	var companyAllRCResponse []getAllRCHandlerForCompanyResponse
+	for _, rc := range allRCMap {
+		isEnrolled := 0
+		if _, ok := enrolledRCMap[rc.ID]; ok {
+			isEnrolled = 1
+		}
+
+		companyAllRCResponse = append(companyAllRCResponse, getAllRCHandlerForCompanyResponse{
```
Comment (1-Harshit) [inline rc/company.rc.go:82]: could have written join but ok 
at least send true false instead of int

**Review by 1-Harshit — CHANGES_REQUESTED** [review]

**Review by 1-Harshit — APPROVED** [review]

**Pull request merged**